aspnetcore/mvc/views/tag-helpers/built-in/samples/TagHelpersBuiltInAspNetCore/Views/Home/Index.cshtml

```diff
@@ -0,0 +1,166 @@
+﻿@{
+    ViewData["Title"] = "Home Page";
+}
+
+<table class="table table-hover">
+    <thead>
+        <tr>
+            <th>Attribute</th>
+            <th>Markup</th>
+            <th>Result</th>
+        </tr>
+    </thead>
+    <tbody>
+        <tr>
+            <td>asp-action</td>
+            <td>
+                <code>
+                    @Html.Raw(Html.Encode(@"<a asp-controller=""Speaker"" asp-action=""Evaluations"">Speaker Evaluations</a>"))
+                </code>
+            </td>
+            <td>
+                <!-- <snippet_AspAction> -->
+                <a asp-controller="Speaker"
+                   asp-action="Evaluations">Speaker Evaluations</a>
+                <!-- </snippet_AspAction> -->
+            </td>
+        </tr>
+        <tr>
+            <td>asp-all-route-data</td>
+            <td>
+                <code>
+                    @Html.Raw(Html.Encode(@"<a asp-route=""speakerevalscurrent"" asp-all-route-data=""parms"">Speaker Evaluations</a>"))
+                </code>
+            </td>
+            <td>
+                <!-- <snippet_AspAllRouteData> -->
+                @{
+                var parms = new Dictionary<string, string>
+                            {
+                                { "speakerId", "11" },
+                                { "currentYear", "true" }
+                            };
+                }
+
+                <a asp-route="speakerevalscurrent"
+                   asp-all-route-data="parms">Speaker Evaluations</a>
+                <!-- </snippet_AspAllRouteData> -->
+            </td>
+        </tr>
+        <tr>
+            <td>asp-area</td>
+            <td>
+                <code>
+                    @Html.Raw(Html.Encode(@"<a asp-area=""Blogs"" asp-controller=""Home"" asp-action=""AboutBlog"">About Blog"))
+                </code>
+            </td>
+            <td>
+                <!-- <snippet_AspArea> -->
+                <a asp-area="Blogs"
+                   asp-controller="Home"
+                   asp-action="AboutBlog">About Blog</a>
+                <!-- </snippet_AspArea> -->
+            </td>
+        </tr>
+        <tr>
+            <td>asp-controller</td>
+            <td>
+                <code>
+                    @Html.Raw(Html.Encode(@"<a asp-controller=""Speaker"" asp-action=""Index"">All Speakers</a>"))
+                </code>
+            </td>
+            <td>
+                <!-- <snippet_AspController> -->
+                <a asp-controller="Speaker"
+                   asp-action="Index">All Speakers</a>
```

Comment:
Odd to separate this by line. Same comment elsewhere.

Reply:
That's only done to improve the display in the docs. On smaller viewport sizes, it's best to avoid the horizontal scrolling.